server/src/main/java/org/apache/druid/client/selector/CustomTierSelectorStrategyConfig.java

```diff
@@ -31,8 +31,16 @@
   @JsonProperty
   private List<Integer> priorities = new ArrayList<>();
 
+  @JsonProperty
+  private List<String> allowedTiers = new ArrayList<>();
```

Comment:
this name should be same as what comes after `druid.broker.select.tier.custom.`. so if you are using `druid.broker.select.tier.custom.tierNames`, then this field should be named `tierNames` or you should have the annotation as `@JsonProperty("tierNames")`
Please add a test for serde if not added already.

Reply:
I had changed these names. Must have missed changing here. Thanks!